File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int curr_max = -1000000000;
        int ans = -1000000000;
        for (int i = 0 ; i < values.length;i++){
            ans = Math.max(ans,values[i] - i + curr_max);
            curr_max = Math.max(curr_max,values[i] + i);
        }
        return ans;
        
    }
}

File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, v: List[int]) -> int:
        ans = - float("inf") ; curr_max = - float("inf")
        for idx , item in enumerate(v):
            val = item - idx
            ans = max(ans,curr_max + val)
            curr_max = max(curr_max,item + idx)
        return ans
            
        
        
            
        